unordered associative task (1)/student/src/impl.cpp

```diff
@@ -1 +1,21 @@
 #include "impl.h"
+#include <unordered_set>
+
+
+/**
+ * @todo Implement function that will check if one string has the same characters
+ * as the second one
+ *
+ * @example is_anagram("abcd", "acdb") -> true
+ * @example is_anagram("abcd", "abec") -> false
+ *
+ * @param first - first word
+ * @param second - second word
+ * @return true if first is anagram of second
+ */
+bool is_anagram(const std::string& first, const std::string& second)
+{
+    std::unordered_multiset<char> firstWord(first.begin(), first.end());
```

Comment:
In such case, ordered multiset will be much simpler to understand.

Smart, but hard to understand and non traditional way :)

Reply:
Will work for chars, but not for the complex objects, because order is unknown

Reply:
такое, таска про знакомство с хеш таблицами, а не про найдите лучший способ решения, или найдите универсальный лучший способ, даны конкретные условия: хеш контейнеры и обычная строка,
тоже самое в таске про деку, почему дека? потому что таска про деки а не про векторы.

здесь unordered_multiset, а не unordered_set для учета повтора букв
и да я в курсе что для произвольного класса надо было бы хеш функцию предоставлять и компоратор

Reply:
Actually, this task will be moved to ordered associative containers section, because expected solution is too unobvious - char counting instead of placing chars actually.
It requires less memory, but...too much code and bit confusing.
That's my fail, sorry :(